src/main/java/com/easytestit/generatexml/service/ReportService.java

```diff
@@ -0,0 +1,486 @@
+            if (featureTagsMap.get("elementTags") != null) tagStrValues += featureTagsMap.get("elementTags");
+            if (featureTagsMap.get("featureTags") != null) tagStrValues += featureTagsMap.get("featureTags");
+
+            tests.put(element.getLine(), new TemporaryTestCase()
+                    .setLine(element.getLine())
+                    .setStatus(testCaseStatus.contains(GenerateXMLConstants.FAILED) ? GenerateXMLConstants.FAILED : GenerateXMLConstants.PASSED)
+                    .setKeyword(element.getKeyword())
+                    .setName(element.getName())
+                    .setDescription(feature.getDescription())
+                    .setTestOutputString(stepOutResults)
+                    .setTestErrorOutputString((stepErrResults != null && !stepErrResults.isEmpty()) ? "Stack Trace: \n".concat(stepErrResults) : null)
+                    .setTestDuration(durationOfTest)
+                    .setTags(tagStrValues));
+
+            durationOfAllTest += durationOfTest;
+            stepOutResults = "";
+            stepErrResults = "";
+            testCaseStatus = "";
+            durationOfTest = 0L;
+
+            featureTagsMap.remove("elementTags");
+        } else {
+            throw new GenerateXMLReportException("Arguments feature & element should not be null but are null. See detailed stack trace: ", new NullPointerException());
+        }
+    }
+
+    private SingleReportSuite getSingleReportSuite(final Feature feature, final Collection<Feature> features) {
+        int increment = 1;
+        int ignored = 0;
+        int disabled = 0;
+        if (this.featureTags.contains("ignored") || this.featureTags.contains("disabled")) {
+            ignored = StringUtils.countMatches(this.featureTags, "ignored");
+            disabled = StringUtils.countMatches(this.featureTags, "disabled");
+        }
+
+        if (this.tests != null) {
+            Collection<TestCase> testCases = this.getTestCasesFromFeature(this.tests);
+            return (
+                    new SingleReportSuite())
+                    .setName((String)this.getLastElement((Iterable)Arrays.stream(feature.getName().split("/")).collect(Collectors.toList())))
+                    .setTests(String.valueOf(this.countScenariosInSuite))
+                    .setFailures(String.valueOf(this.countFailuresTestFromOneFile))
+                    .setDisabled(String.valueOf(ignored + disabled))
+                    .setHostname(this.hostName)
+                    .setId(String.valueOf(((ArrayList)features).indexOf(feature) + increment))
+                    .setPackageName(feature.getName())
+                    .setSkipped(String.valueOf(this.countSkippedTestFromOneFile))
+                    .setTime(String.valueOf(UtilsConverter.round.apply((double)this.durationOfAllTest * 1.0E-9D)))
+                    .setTimestamp(LocalDateTime.parse(this.responseDate, DateTimeFormatter.ofPattern("EEE, d MMM yyyy HH:mm:ss", Locale.ENGLISH)).toString())
```

Comment:
It is a place where this.responseDate can be without a correct value. As suggestion, here can be try-catch block for date parsing and log if fails date parsing

Reply:
Yeah, I remember your notice. Can you create after this pull request an issue and make changes in the new branch with the name this issue, and then through the pool request, etc.?

Reply:
Ok. After PR will create an issue and fix in a separate branch